packages/teams-js/src/public/media.ts

```diff
@@ -431,11 +434,31 @@ export namespace media {
     }
 
     /**
+     * Function to programatically stop the ongoing media event
+     *
+     * @returns A resolved promise
+     * */
+    public stop(): Promise<void>;
+    /**
+     * @deprecated
+     * As of 2.0.0-beta.3, please use {@link MediaController.stop MediaController.stop(): Promise\<void\>} instead.
+     
      * Function to programatically stop the ongoing media event
      * Optional; @param callback is used to send app if host client has successfully stopped the event or not
      */
-    public stop(callback?: (err?: SdkError) => void): void {
-      this.notifyEventToHost(MediaControllerEvent.StopRecording, callback);
+    public stop(callback: (err?: SdkError) => void): void;
+    public stop(callback?: (err?: SdkError) => void): Promise<void> {
+      if (callback) {
+        // if there is a callback we want to pass the callback to the notifyevent to host fn - if an exception is caught, it gets passed to the callback
+        const inputFn = (): Promise<void> =>
+          new Promise<void>(resolve => {
+            resolve(this.notifyEventToHost(MediaControllerEvent.StartRecording, callback));
+          });
+        return callCallbackWithErrorOrResultFromPromiseAndReturnPromise(inputFn, callback);
```

Comment:
Most of the other instances in our code base like this have one call to callCallbackWithErrorOrResultFromPromiseAndReturnPromise at the end of the function and then they pass the wrapped function and the callback to that. Is there a reason that that's not the pattern we're following here? I think callCallbackWithErrorOrResultFromPromiseAndReturnPromise has logic for handling the if (callback) scenario

Reply:
Curious to hear why you decided not to go with this pattern? I don't think it's wrong necessarily, but does mean that we're not following the same pattern as a lot of the other backcompatible public functions.